be2-scala/src/main/scala/ch/epfl/pop/pubsub/graph/validators/RollCallValidator.scala

```diff
@@ -34,38 +38,33 @@ case object RollCallValidator extends MessageDataContentValidator {
     }
   }
 
-  // TODO check that this is correct (correct validations)
-  def validateOpenRollCall(rpcMessage: JsonRpcRequest): GraphMessage = {
-    def validationError(reason: String): PipelineError = super.validationError(reason, "OpenRollCall", rpcMessage.id)
+  def validateOpenRollCall(rpcMessage: JsonRpcRequest, validatorName: String = "OpenRollCall"): GraphMessage = {
+    def validationError(reason: String): PipelineError = super.validationError(reason, validatorName, rpcMessage.id)
 
     rpcMessage.getParamsMessage match {
       case Some(message: Message) =>
         val data: OpenRollCall = message.decodedData.get.asInstanceOf[OpenRollCall]
 
-        if (!validateTimestampStaleness(data.start)) {
-          Right(validationError(s"stale 'start' timestamp (${data.start})"))
+        val laoId: Hash = rpcMessage.extractLaoId
+        // Note: this id doesnt follow the standard sh256(string, hash, timestamp, string) other events do;
+        // this is why generateValidationId cannot be used
+        val expectedRollCallId: Hash = Hash.fromStrings(
+          EVENT_HASH_PREFIX, laoId.toString, data.opens.toString, data.opened_at.toString
+        )
+
+        if (!validateTimestampStaleness(data.opened_at)) {
+          Right(validationError(s"stale 'opened_at' timestamp (${data.opened_at})"))
+        } else if (expectedRollCallId != data.opens) {
```

Comment:
We're comparing the wrong item here.
the `opens` id should be in the database, but it's `update_id` which should be `== expectedRollCallId`
cf. https://github.com/dedis/student_21_pop/blob/master/fe1-web/model/network/method/message/data/rollCall/OpenRollCall.ts